cql3/dialect.hh

```diff
@@ -0,0 +1,12 @@
+// Copyright (C) 2024-present ScyllaDB
+// SPDX-License-Identifier: AGPL-3.0-or-later
+
+#pragma once
+
+namespace cql3 {
+
+struct dialect {
+    bool duplicate_bind_variable_names_refer_to_same_variable = true;  // if :a is found twice in a query, the two references are to the same variable
```

Comment:
Perhaps it will be useful to refer to the issue number, or test, or something, that expands on this difference. And/or mention that "false" is the behavior of Scylla 5.0 (IIRC) and earlier, and all Cassandra, and "true" is Scylla's default since 5.1 (IIRC).
I think without an expanded comment, a year (or a week ;-)) from now, nobody will remember why this flag exists. I guess "git blame" could help, but as the code moves around it becomes harder to follow the (hi)story of this flag.